tools/addrundata/case.go

```diff
@@ -17,31 +16,26 @@ import (
 
 // testcase carries parsed rundata from different sources to be fixed and checked.
 type testcase struct {
-	pkg      string        // Package used in the test code.
-	markdown *mpb.Metadata // From the README.md.
-	existing *mpb.Metadata // From existing source code.
-	fixed    *mpb.Metadata // Fixed rundata to write back, populated by fix().
+	markdown   *mpb.Metadata // From the README.md.
+	existing   *mpb.Metadata // From existing source code.
+	deprecated bool          // Whether a deprecated rundata_test.go file was found.
+	fixed      *mpb.Metadata // Fixed rundata to write back, populated by fix().
 }
 
 // read reads the markdown and existing rundata from the test directory.
 func (tc *testcase) read(testdir string) error {
 	if err := readFile(filepath.Join(testdir, "README.md"), tc.readMarkdown); err != nil {
 		return fmt.Errorf("could not parse README.md: %w", err)
 	}
-	testpaths, err := filepath.Glob(filepath.Join(testdir, "*_test.go"))
-	if err != nil {
-		return fmt.Errorf("could not glob: %w", err)
-	}
-	for _, testpath := range testpaths {
-		if err := readFile(testpath, tc.readPackage); err != nil {
-			return fmt.Errorf("could not detect test package: %w", err)
-		}
-		if tc.pkg != "" {
-			break
+	if err := readFile(filepath.Join(testdir, "rundata_test.go"), tc.readCode); err != nil {
```

Comment:
Suggest use switch to avoid double nesting?

```go
switch err := readFile(...) {
case err == nil:
  tc.deprecated = true
case os.IsNotExist(err):
  // This is fine.
default:
  return fmt.Errorf(...)
}

Reply:
Done